runtime/src/main/java/com/dylibso/chicory/runtime/Store.java

```diff
@@ -129,8 +130,16 @@ public Store register(String name, Instance instance) {
      * A shorthand for instantiating a module and registering it in the store.
      */
     public Instance instantiate(String name, WasmModule m) {
+        return this.instantiate(
+                name, imports -> Instance.builder(m).withImportValues(imports).build());
+    }
+
+    /**
+     * A shorthand for instantiating a module and registering it in the store.
+     */
+    public Instance instantiate(String name, Function<ImportValues, Instance> instanceBuilder) {
```

Comment:
you could pass directly an `Instance.Builder` though

Reply:
I entertained the idea, and I think that giving max freedom to the user is the way to go to avoid wrapping stuff.

Reply:
maybe let's rename the var from `instanceBuilder` to `instanceFactory` to avoid confusion.

Reply:
done :+1: